src/SprykerEco/Zed/Adyen/Business/Handler/Notification/AdyenNotificationHandler.php

```diff
@@ -0,0 +1,118 @@
+    protected $utilEncodingService;
+
+    /**
+     * @var \SprykerEco\Zed\Adyen\Business\Reader\AdyenReaderInterface
+     */
+    protected $reader;
+
+    /**
+     * @var \SprykerEco\Zed\Adyen\Business\Writer\AdyenWriterInterface
+     */
+    protected $writer;
+
+    /**
+     * @var \SprykerEco\Zed\Adyen\AdyenConfig
+     */
+    protected $config;
+
+    /**
+     * @param \SprykerEco\Zed\Adyen\Dependency\Service\AdyenToUtilEncodingServiceInterface $utilEncodingService
+     * @param \SprykerEco\Zed\Adyen\Business\Reader\AdyenReaderInterface $reader
+     * @param \SprykerEco\Zed\Adyen\Business\Writer\AdyenWriterInterface $writer
+     * @param \SprykerEco\Zed\Adyen\AdyenConfig $config
+     */
+    public function __construct(
+        AdyenToUtilEncodingServiceInterface $utilEncodingService,
+        AdyenReaderInterface $reader,
+        AdyenWriterInterface $writer,
+        AdyenConfig $config
+    ) {
+        $this->utilEncodingService = $utilEncodingService;
+        $this->reader = $reader;
+        $this->writer = $writer;
+        $this->config = $config;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\AdyenNotificationsTransfer $notificationsTransfer
+     *
+     * @return \Generated\Shared\Transfer\AdyenNotificationsTransfer
+     */
+    public function handle(AdyenNotificationsTransfer $notificationsTransfer): AdyenNotificationsTransfer
+    {
+        $isSuccess = true;
+
+        foreach ($notificationsTransfer->getNotificationItems() as $notificationItem) {
+            try {
+                $this->handleNotification($notificationItem);
+            } catch (AdyenNotificationNotHandledException $exception) {
+                $isSuccess = false;
+                continue;
+            }
+        }
+
+        $notificationsTransfer->setIsSuccess($isSuccess);
+
+        return $notificationsTransfer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\AdyenNotificationRequestItemTransfer $notificationTransfer
+     *
+     * @throws \SprykerEco\Zed\Adyen\Business\Exception\AdyenNotificationNotHandledException
+     *
+     * @return void
+     */
+    protected function handleNotification(AdyenNotificationRequestItemTransfer $notificationTransfer): void
+    {
+        if (!$notificationTransfer->getSuccess()) {
+            throw new AdyenNotificationNotHandledException('Notification Request Item is not success.');
+        }
+
+        $statuses = $this->config->getMappedOmsStatuses();
+        if (!array_key_exists($notificationTransfer->getEventCode(), $statuses)) {
+            throw new AdyenNotificationNotHandledException('Event code does not registered.');
+        }
+
+        $paymentAdyenTransfer = $this->reader->getPaymentAdyenByPspReference($notificationTransfer->getPspReference());
+        if (!$paymentAdyenTransfer->getFkSalesOrder()) {
+            throw new AdyenNotificationNotHandledException(
+                'Payment entity with requested pspReference does not registered.'
```

Comment:
'Payment entity with requested pspReference **IS** not registered.'

You want a const for this.